### PR TITLE
🔧 CI: add code quality enforcement strategies

### PR DESCRIPTION
**Pull request by AlexAndradeNet**
### The main changes

- [📝 add documentation](https://github.com/CarlosEstradaQA/Cypress_Automation/commit/9522295b449d0120566af3398f6692d592a39186).
- [🔨 add sonarqube support](https://github.com/CarlosEstradaQA/Cypress_Automation/commit/8a777a90720f8f37c87c992fd330010fbfdc6d27).
- [🔨 add linter tools](https://github.com/CarlosEstradaQA/Cypress_Automation/commit/a47d08f7426da0af7e2ba4e5c7f94ade214ace17).
- [🔧 add config files for IDE code quality enforcement during development time](https://github.com/CarlosEstradaQA/Cypress_Automation/commit/5c0b4e37e9729f291b911419f264a1bfca1d914e).
- [🔨 add Husky scripts](https://github.com/CarlosEstradaQA/Cypress_Automation/commit/152d594067e664bbc10cbff46e557e8b5a9c23f7).

### Immediate next actions

1. @CarlosEstradaQA: Configure GitHub actions and Gitflow practices as indicated in [this document](https://blankfactor.atlassian.net/wiki/spaces/QA/pages/937000971/GitHub+Actions).
2. @CarlosEstradaQA: Lint all files with the following command:
  ```sh
   npm run npm run lintFixAll
  ```
3. @CarlosEstradaQA: Send a PR with changes.

### Future actions
- @AlexAndradeNet: Configure GitHub actions.
- @CarlosEstradaQA: Fix all SonarQube recommendations. See file [CONTRIBUTE.md](https://github.com/CarlosEstradaQA/Cypress_Automation/blob/alex/code-quality-enforcement/CONTRIBUTE.md).

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "root": true,
+    "env": {
+        "browser": true,
+        "node": true,
+        "cypress/globals": true
+    },
+    "extends": ["airbnb", "eslint:recommended", "plugin:cypress/recommended"],
```
Comment (CarlosEstradaQA) [inline .eslintrc.json:8]: Is this necessary?

File: .gitignore [hunk 2]
```diff
@@ -36,4 +35,16 @@ yarn-error.log*
...
 # Ignore all local history of files
 .history
-.ionide
\ No newline at end of file
+.ionide
+
+# Idea IDE
+# IntelliJ IDEA files
+# Ignore everything in the folder .idea except .idea/externalDependencies.xml
+**/.idea/*
```
Comment (CarlosEstradaQA) [inline .gitignore:44]: Are you sure that IntelliJ IDEA is a necessary or good approach to run this project?

File: .prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "trailingComma": "es5",
```
Comment (CarlosEstradaQA) [inline .prettierrc.json:2]: It's better to use ES6 instead of ES5, which provides more features for JavaScript and has more lifetime of browser support.

**Review by CarlosEstradaQA — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (AlexAndradeNet) [reply .gitignore:44]: I improved the commentary to make it clear that this applies to all Jbrains IDEs, in this case, WebStorm for Javascript.

File: .prettierrc.json (hunk 3 above)
Comment (AlexAndradeNet) [reply .prettierrc.json:2]: You are right, but currently, they have not implemented "trailingComma" for ES6, or the new ES6 standard has not implemented changes to this parameter. The following is the link to the implementation https://github.com/prettier/prettier/tree/main/tests/format/js/trailing-comma

File: .eslintrc.json (hunk 1 above)
Comment (AlexAndradeNet) [reply .eslintrc.json:8]: Yes, this is necessary to make Prettier and Eslint work as a team; with this configuration, you could get the best of both worlds.

In this case, they use the following order:
- Apply Cypress code style guidelines, just for Cypress stuff.
- Apply [AirBnB Javascript code style guidelines](https://medium.com/@corentinjacques/airbnbs-javascript-style-guide-5fe9100d2386) for the rest of the Javascript code.
- Apply Eslint's recommendation for those the previous ones didn't have a rule.
- Default: apply Prettier code style guidelines.

**Review by CarlosEstradaQA — APPROVED** [review]

**Pull request merged**